ProgLabHW4/mainHW4.py

```diff
@@ -0,0 +1,259 @@
+from Validation import *
+from Collection import Collection
+from Event import Event
+from RestaurantNames import RestaurantName
+import datetime
+
+
+def menu():
+    print('\nA to add')
+    print('D to delete')
+    print('E to edit')
+    print('S to sort')
+    print('F to search')
+    print('P to show all')
+    print('Q to quit\n')
+
+
+def edit_sort_menu():
+    print('\nT - Title')
+    print('R - Restaurant name')
+    print('Da - Date')
+    print('Ti - Starting time')
+    print('D - Duration')
+    print('P - Price\n')
+
+
+def searching_by(parameter):
+    if parameter == 'T':
+        print('Sorting by titles')
+    elif parameter == 'Da':
+        print('Sorting by dates')
+    elif parameter == 'Ti':
+        print('Sorting by starting times')
+    elif parameter == 'R':
+        print('Sorting by restaurant names')
+    elif parameter == 'D':
+        print('Sorting by event durations')
+    elif parameter == 'P':
+        print('Sorting by ticket price')
+
+
+def restaurants():
+    for q in RestaurantName:
+        print(q.value, '-', q.name)
+
+
+def delete_from_collection(to_be_deleted, collection):
+    if not Validation.validate_int(to_be_deleted):
+        to_be_deleted = int(to_be_deleted) - 1
+        if not Validation.validate_bounds(to_be_deleted, collection):
+            Event.amount += -1
+            return collection.delete_at(to_be_deleted)
+        else:
+            return 'Event was not deleted: no such ID'
+    else:
+        return 'Event was not deleted: ID is not an int'
+
+
+def find_in_collection(to_be_found, collection):
+    if not Validation.validate_int(to_be_found):
+        return collection.search(to_be_found, True, False)
+    elif not Validation.validate_float(to_be_found):
+        return collection.search(to_be_found, False, True)
+    else:
+        return collection.search(to_be_found, False, False)
+
+
+def overwrite(collection):

```

Comment:
1) it had to be in collection
2) check if file exists
3) name of file from console